cmd/sidecars/cloudinit/cloudinit.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 StackPath, LLC
+ *
+ */
+
+// Inspired by cmd/example-hook-sidecar
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/pflag"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	cloudinit "kubevirt.io/kubevirt/pkg/cloud-init"
+)
+
+func preCloudInitIso(log *log.Logger, vmiJSON, cloudInitDataJSON []byte) (string, error) {
+	log.Print("Hook's PreCloudInitIso callback method has been called")
+
+	vmi := v1.VirtualMachineInstance{}
+	err := json.Unmarshal(vmiJSON, &vmi)
+	if err != nil {
+		return "", fmt.Errorf("Failed to unmarshal given VMI spec: %s", vmiJSON)
```

Comment:
nit: would it be useful to print also the actual error message? wdyt?
```suggestion
		return "", fmt.Errorf("Failed to unmarshal given VMI spec: %v %s", err, vmiJSON)
```
Same with the others `fmt.Errorf(...)` in this file and in `diskimage.go`.

Reply:
Added this commit https://github.com/kubevirt/kubevirt/pull/10320/commits/8880313e4682f871ca0099517eaef72a26a77097 for `smbios` sidecar because it was missing already this fix from when I moved it to use `sidecar-shim` in d6950fb292b72f3aa9a291a264f76fadfef5fcb9. With both `cloudinit`  and `disk-mutation` sidecars, the diff is very similar  in their *use sidecar-shim* commits. Thanks for the review!